### PR TITLE
#605 partly fixed, but still issues

### PR DESCRIPTION
**Pull request by githubertus**
sorry, we had a bad side effect with the first fix
1. moved ossn_unregister_menu from line 150 into "if type == 'comments:entity' condition
because otherwise 'delete menus' are removed from type == 'comments:posts'
2. added || (ossn_loggedin_user()->guid == $comment->owner_guid) to type == 'comments:posts'
otherwise we can't delete our own comment from a posting which is owned by someone else

### CODE REVIEW

**Comment by githubertus** [comment]
Stop, don't merge! There's still something wrong

**Comment by lianglee** [comment]
I have tested and your first pull request works fine, i just refactored it : https://github.com/opensource-socialnetwork/opensource-socialnetwork/commit/91375e7a5f270555b42d1ee1d4e702ce23383b13

$params is not actually supplied with wrong ids. ossn_comment_menu is a callback, it is called for each comment item. The issue was, since condition is false in the last comment wasn't registering a menu item but it was using old menu item parameters. 

    if (($user->guid == $params['owner_guid']) || ossn_isAdminLoggedin() 
		|| ($comment->type == 'comments:entity' && $entity->type = 'user' && $user->guid == $entity->owner_guid)) {

     
    } else {
     //unregistered old menu item parameters
   }

In other word once callback is triggered each callback contained previous callback menu item settings , if current menu item didn't register any menu item, its uses old menu item settings which is wrong. We have simply unregistered old menu item settings.

**Comment by lianglee** [comment]
Try to test on latest build and let me know if it works.

**Comment by lianglee** [comment]
Lol let me try to explain again:

comment 35 from admin : param-id = 35
comment 36 from admin : param-id = 36

[
comment 37 from micha : param-id still 36  <= Callback for this comment isn't register nor unregister any comment. So its using menu item settings from previous comments 36

comment 40 from micha : param-id still 36  ] <= Same for this , its using settings from 36 as in both comments there is no delete menu item.

**Comment by githubertus** [comment]
I just tested your build from today: No more problems with wrong ids. That's the good news, but:
posting-delete-prob

as a side effect you can not delete other member's comments from your own posting.

The explanation: first, the delete menu is created correctly in the upper 'posting' part of the code. But then, the entity permission check is made (although no entity) and since every conditions results to be FALSE here if you're not the owner of the comment, too, your new else case is reached and the just created delete menu gets destroyed. ;)

That's why I moved the second permission check completely inside the if($comment->type == 'comments:entity'){ condition in my pull request from yesterday. Either we have a post type, or an entity type.

**Comment by lianglee** [comment]
Will fix this, yes i saw this ;) https://github.com/opensource-socialnetwork/opensource-socialnetwork/issues/607
